docs/source/data/errors.mdx

```diff
@@ -604,21 +604,60 @@ In this case, the error above is reported to Apollo Studio as:
 The REDACTED doesn't have sufficient privileges.
 ```
 
-## Returning HTTP status codes
+## Setting HTTP status code and headers
 
 GraphQL, by design, does not use the same conventions from REST to communicate via HTTP verbs and status codes. Client information should be contained in the schema or as part of the standard response `errors` field. We recommend using the included [Error Codes](#built-in-error-codes) or [Custom Errors](#custom-errors) for error consistency rather than directly modifying the HTTP response.
 
-You can set custom fields on your HTTP response by using a [plugin](/apollo-server/integrations/plugins). Be aware that GraphQL client libraries may not treat all response status codes the same, and so it will be up to your team to decide what patterns to use.
+Apollo Server's general behavior with respect to status codes is as follows:
+- If Apollo Server has not properly started up or is in the process of shutting down, use status code 500. (The former only happens if you're using a serverless integration instead of `await`ing `startStandaloneServer(server)` or `server.start()`; the latter only happens if you have not properly set up draining for your non-standalone integration.)
+- If Apollo Server cannot parse the request into a legal GraphQL document and validate it against your schema, use status code 400. This also includes other problems with the request, such as attempting to send a batched HTTP request when `allowBatchedHttpRequests` is not enabled, or if CSRF prevention blocks the request.
+- If an invalid HTTP method is used (`GET` with a mutation, or any HTTP method other than `GET` or `POST`), use status code 405.
+- If your `context` function throws, use status code 500.
+- If there is an unexpected error during the processing of the request (either a bug in Apollo Server, or a plugin hook that throws), use status code 500.
+- Otherwise, use status code 200. This is essentially the case where the server is able to execute the GraphQL operation, and execution completes successfully (possibly with some resolver-specific errors).
 
-As an example, here is how you could set a custom response header and status code based on a GraphQL error:
+You can affect the HTTP status code or set response headers either by throwing an error in a resolver or `context` function, or by writing a [plugin](/apollo-server/integrations/plugins). Be aware that GraphQL client libraries may not treat all response status codes the same, and so it will be up to your team to decide what patterns to use. We strongly encourage you to avoid setting the HTTP status code based on an error thrown from a resolver; setting the status code from context creation errors can be more useful.
```

Comment:
How would you feel about breaking this out more? Something like this:
```suggestion
There are three ways to change an HTTP status code or set custom response headers. You can: throw an error in a resolver, throw an error in your `context` function, or you can write a [plugin](/apollo-server/integrations/plugins).

Be aware that GraphQL client libraries might not treat all response status codes the same, so it will be up to your team to decide which patterns to use. You should _avoid_ setting HTTP status codes based on errors thrown by resolvers. Instead, we recommend setting the HTTP status code based on `context` creation errors. 
```

Reply:
First paragraph: can the three options be written more parallel?

I like the second paragraph but maybe even write something like "While Apollo Server does allow you to set, HTTP status codes based on errors thrown by resolvers, best practices for GraphQL over HTTP encourage sending 200 whenever the operation executes" or something?

Reply:
I'm sorry, I'm not sure what you mean by more parallel for the first paragraph. Like all in one sentence? 

How would you feel about this in terms of order:
```suggestion
There are three ways to change an HTTP status code or set custom response headers, you can: throw an error in a resolver, throw an error in your `context` function, or write a [plugin](/apollo-server/integrations/plugins).  

While Apollo Server does enable you to set HTTP status codes based on errors thrown by resolvers, best practices for GraphQL over HTTP encourage sending 200 whenever an operation executes.  Instead, we recommend setting the HTTP status code based on `context` creation errors. 

Additionally, be aware that GraphQL client libraries might not treat all response status codes the same, so it will be up to your team to decide which patterns to use. 
```

Reply:
I just meant that you reiterated "you can" in the third option, which you fixed even if you didn't understand me :)

Reply:
```suggestion
There are three ways to change an HTTP status code or set custom response headers, you can: throw an error in a resolver, throw an error in your `context` function, or write a [plugin](/apollo-server/integrations/plugins).

While Apollo Server does enable you to set HTTP status codes based on errors thrown by resolvers, best practices for GraphQL over HTTP encourage sending 200 whenever an operation executes, so we don't recommend you use this mechanism in resolvers, just in the `context` function or in a plugin hooking into an early stage of the request pipeline.

Be aware that GraphQL client libraries might not treat all response status codes the same, so it will be up to your team to decide which patterns to use. 
```

Reply:
What do you think about this wording? I don't want to actively say "you should set the HTTP status code", like, people should do it if they have a reason to but it's fine not to. The note is just encouraging folks to *not* do that in certain cases

Reply:
I love the first two parts; the middle sentence feels a bit long. I think splitting the middle sentence into two makes this flow just right:

```suggestion
There are three ways to change an HTTP status code or set custom response headers, you can: throw an error in a resolver, throw an error in your `context` function, or write a [plugin](/apollo-server/integrations/plugins).

While Apollo Server does enable you to set HTTP status codes based on errors thrown by resolvers, best practices for GraphQL over HTTP encourage sending 200 whenever an operation executes. So, we don't recommend using this mechanism in resolvers, just in the `context` function or in a plugin hooking into an early stage of the request pipeline.

Be aware that GraphQL client libraries might not treat all response status codes the same, so it will be up to your team to decide which patterns to use.
```